### DISCUSSION TITLE
Ebyte E22-900MM22S (SX1262) -707 Error

### DISCUSSION THREAD

**Discussion by SamTheil** [jgromes/RadioLib#1357] [General]
I'm working on a custom esp32-s3-mini based arduino board with an Ebyte E22-900MM22S radio module and I'm having issues interfacing and connecting to the radio module with RadioLib. The Ebyte module claims to operate in accordance with the SX1268/SX1262 chip volume so I programmed it with RadioLib using an sx1262 instance. When I try to initilize the module I always get a -707 error even when setting TCXO to zero (something I read in the troubleshooting guide). It is worth noting that I have not connected the DIO1 pin, I'm assuming this isn't the issue because that pin is only needed for interrupts and not configuring or controlling the device.

I have tried this program on two different esp32 boards and three different Ebyte modules. Would greatly appreciate any help in figuring this out. There is very little information online about how to program the E22-900MM22S so if anyone has general experience getting them working that would be super helpful as well. Thanks!

```cpp
#include <Arduino.h>
#include <SPI.h>
#include <RadioLib.h>

// Pin definitions
#define SPI_MOSI_PIN 11
#define SPI_MISO_PIN 13
#define SPI_SCK_PIN 12
#define CS_PIN 10
#define RESET_PIN 2
#define BUSY_PIN 8
#define RXEN_PIN 7
#define TXEN_PIN 6

// Create a RadioLib SX1262 instance, setting DIO1 to -1 because it is not connected
SX1262 radio = new Module(CS_PIN,-1,RESET_PIN, BUSY_PIN);

void setup() {
    Serial.begin(115200);

    delay(5000);

    Serial.println("Starting SX1262 setup...");

    // Configure TXEN and RXEN pins
    pinMode(TXEN_PIN, OUTPUT);
    pinMode(RXEN_PIN, OUTPUT);
    pinMode(RESET_PIN, OUTPUT);

    digitalWrite(TXEN_PIN, HIGH);
    digitalWrite(RXEN_PIN, LOW);
    delay(10);

    digitalWrite(RESET_PIN, LOW);
    delay(200);
    digitalWrite(RESET_PIN, HIGH);
    delay(200);

    SPI.begin(SPI_SCK_PIN, SPI_MISO_PIN, SPI_MOSI_PIN, CS_PIN);

    // initialization failed with and without setting TCXO to zero
    radio.setTCXO(0);

    // Initialize the SX1262 radio module
    int state = radio.begin();
    if (state == RADIOLIB_ERR_NONE) {
        Serial.println("SX1262 initialized successfully.");
    } else {
        Serial.print("Failed to initialize SX1262. Error code: ");
        Serial.println(state);
        while (true);
    }
}

void loop() {
}
```

**Comment by jgromes** [comment]
You need to to either set the TCXO voltage in the begin method, or use `radio.XTAL = true;` before calling begin. `radio.setTCXO` won't work before the module is initialzied with `begin`, which already needs to have the correct TCXO voltage otherwise the init will fail - a bit of a chicken-or-egg problem.

**Reply by SamTheil** [reply]
That worked, thanks! Just realized that you had mentioned this in the troubleshooting guide as well so apologies for bringing it up again. Appreciate the quick response!
